.changeset/sour-gorillas-fail.md

```diff
@@ -0,0 +1,23 @@
+---
+'@backstage/plugin-scaffolder-backend': patch
+---
+
+# Repo visibility for GitLab and BitBucket repos
+
+**NOTE: This changes default repo visibility from `private` to `public` for GitLab and BitBucket** which
+is consistent with the GitHub default. If you were counting on `private` visibility, you'll need to update
+your scaffolder config to use `private`.
+
+This adds repo visibility feature parity with GitHub for GitLab and BitBucket.
+
+To configure the repo visibility, set scaffolder._type_.visibility as in this example:
+
+```yaml
+scaffolder:
+  github:
+    visibility: private # 'public' or 'internal' or 'private' (default is 'public')
+  gitlab:
+    visibility: public # 'public' or 'internal' or 'private' (default is 'public')
+  bitbucket:
+    visibility: public # 'public' or 'private' (default is 'public')
```

Comment:
ooof i see the schema for the scaffolder leaves some things to be desired, but anyway, these should probably be added there https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/config.d.ts#L18

Reply:
I don't quite understand what this entry for github is about:
```
      [key: string]: string;
```
But will give it a try

Reply:
Yeah scaffolder schema does not support the integrations concepts of a list of hosts for each flavor, so things like `visibility` become global for all instances of a given remote repo flavor.

Reply:
Got that wrong. Standby.

Reply:
Let's try that again. Also rebased to 5df0714

Reply:
Yeah it's just a "match anything here" basically; at the time it was written somebody took a quick solution without defining the full field details :)